pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -258,435 +259,447 @@ private BrokerResponseNative handleRequest(long requestId, String query,
+          realtimeTableName = tableName;
+        }
+      } else {
+        // Hybrid table (check both OFFLINE and REALTIME)
+        String offlineTableNameToCheck = TableNameBuilder.OFFLINE.tableNameWithType(tableName);
+        if (_routingManager.routingExists(offlineTableNameToCheck)) {
+          offlineTableName = offlineTableNameToCheck;
+        }
+        String realtimeTableNameToCheck = TableNameBuilder.REALTIME.tableNameWithType(tableName);
+        if (_routingManager.routingExists(realtimeTableNameToCheck)) {
+          realtimeTableName = realtimeTableNameToCheck;
+        }
+      }
 
-    if (offlineTableName == null && realtimeTableName == null) {
-      // No table matches the request
-      if (realtimeTableConfig == null && offlineTableConfig == null) {
-        LOGGER.info("Table not found for request {}: {}", requestId, query);
-        requestContext.setErrorCode(QueryException.TABLE_DOES_NOT_EXIST_ERROR_CODE);
-        return BrokerResponseNative.TABLE_DOES_NOT_EXIST;
+      TableConfig offlineTableConfig =
+          _tableCache.getTableConfig(TableNameBuilder.OFFLINE.tableNameWithType(rawTableName));
+      TableConfig realtimeTableConfig =
+          _tableCache.getTableConfig(TableNameBuilder.REALTIME.tableNameWithType(rawTableName));
+
+      if (offlineTableName == null && realtimeTableName == null) {
+        // No table matches the request
+        if (realtimeTableConfig == null && offlineTableConfig == null) {
+          LOGGER.info("Table not found for request {}: {}", requestId, query);
+          requestContext.setErrorCode(QueryException.TABLE_DOES_NOT_EXIST_ERROR_CODE);
+          return BrokerResponseNative.TABLE_DOES_NOT_EXIST;
+        }
+        LOGGER.info("No table matches for request {}: {}", requestId, query);
+        requestContext.setErrorCode(QueryException.BROKER_RESOURCE_MISSING_ERROR_CODE);
+        _brokerMetrics.addMeteredGlobalValue(BrokerMeter.RESOURCE_MISSING_EXCEPTIONS, 1);
+        return BrokerResponseNative.NO_TABLE_RESULT;
       }
-      LOGGER.info("No table matches for request {}: {}", requestId, query);
-      requestContext.setErrorCode(QueryException.BROKER_RESOURCE_MISSING_ERROR_CODE);
-      _brokerMetrics.addMeteredGlobalValue(BrokerMeter.RESOURCE_MISSING_EXCEPTIONS, 1);
-      return BrokerResponseNative.NO_TABLE_RESULT;
-    }
 
-    // Handle query rewrite that can be overridden by the table configs
-    if (offlineTableName == null) {
-      offlineTableConfig = null;
-    }
-    if (realtimeTableName == null) {
-      realtimeTableConfig = null;
-    }
-    HandlerContext handlerContext = getHandlerContext(offlineTableConfig, realtimeTableConfig);
-    if (handlerContext._disableGroovy) {
-      rejectGroovyQuery(serverPinotQuery);
-    }
-    if (handlerContext._useApproximateFunction) {
-      handleApproximateFunctionOverride(serverPinotQuery);
-    }
+      // Handle query rewrite that can be overridden by the table configs
```

Comment:
(nit) you may want to check formatting of changes in this class. Seems weird

Reply:
You can probably view this class using the omit whitespace change mode. It show correctly in intellij, I guess some bug with github default diff view.